`supabase db start` error: role "pgbouncer" already exists (SQLSTATE 42710)

# Bug report

## Describe the bug

I'm getting this error when running `supabase db start` in CI (using `supabase/setup-cli`):
```
[...]
bd676e6cd7b2: Pull complete
Digest: sha256:5407d1da475c2ea8c87dec6069ef046995b26b19a7baf97a2f8feae5ea3cb605
Status: Downloaded newer image for ghcr.io/supabase/postgres:14.1.0.89
Error: ERROR: role "pgbouncer" already exists (SQLSTATE 42710)
At statement 11: CREATE ROLE pgbouncer
Try rerunning the command with --debug to troubleshoot the error.
Error: Process completed with exit code 1.
```

My SQL migrations have nothing mentioning `pgbouncer`. It started happening without any schema changes

This started in `v1.42.3`

## To Reproduce

Run `supabase db start`

## System information

- OS: GitHub action `ubuntu-latest`

--- a/internal/db/diff/migra.go
+++ b/internal/db/diff/migra.go
@@ -122,7 +122,11 @@ func likeEscapeSchema(schemas []string) (result []string) {
 func CreateShadowDatabase(ctx context.Context) (string, error) {
 	config := container.Config{
 		Image: utils.DbImage,
-		Env:   []string{"POSTGRES_PASSWORD=postgres"},
+		Env: []string{
+			"POSTGRES_PASSWORD=postgres",
+			"POSTGRES_HOST=/var/run/postgresql",
+			"POSTGRES_INITDB_ARGS=--lc-ctype=C.UTF-8",
+		},
 	}
 	if utils.Config.Db.MajorVersion >= 14 {
 		config.Cmd = []string{"postgres",
@@ -134,7 +138,7 @@ func CreateShadowDatabase(ctx context.Context) (string, error) {
 	hostPort := strconv.FormatUint(uint64(utils.Config.Db.ShadowPort), 10)
 	hostConfig := container.HostConfig{
 		PortBindings: nat.PortMap{"5432/tcp": []nat.PortBinding{{HostPort: hostPort}}},
-		Binds:        []string{"/dev/null:/docker-entrypoint-initdb.d/migrate.sh:ro"},
+		Tmpfs:        map[string]string{"/docker-entrypoint-initdb.d": ""},
 		AutoRemove:   true,
 	}
 	return utils.DockerStart(ctx, config, hostConfig, "")

--- a/internal/db/start/start.go
+++ b/internal/db/start/start.go
@@ -45,7 +45,7 @@ func StartDatabase(ctx context.Context, fsys afero.Fs, w io.Writer, options ...f
 		Env: []string{
 			"POSTGRES_PASSWORD=postgres",
 			"POSTGRES_HOST=/var/run/postgresql",
-			"LC_ALL=C.UTF-8",
+			"POSTGRES_INITDB_ARGS=--lc-ctype=C.UTF-8",
 		},
 		Healthcheck: &container.HealthConfig{
 			Test:     []string{"CMD", "pg_isready", "-U", "postgres", "-h", "localhost", "-p", "5432"},
@@ -74,10 +74,8 @@ func StartDatabase(ctx context.Context, fsys afero.Fs, w io.Writer, options ...f
 	hostConfig := container.HostConfig{
 		PortBindings:  nat.PortMap{"5432/tcp": []nat.PortBinding{{HostPort: hostPort}}},
 		RestartPolicy: container.RestartPolicy{Name: "always"},
-		Binds: []string{
-			utils.DbId + ":/var/lib/postgresql/data",
-			"/dev/null:/docker-entrypoint-initdb.d/migrate.sh:ro",
-		},
+		Binds:         []string{utils.DbId + ":/var/lib/postgresql/data"},
+		Tmpfs:         map[string]string{"/docker-entrypoint-initdb.d": ""},
 	}
 	fmt.Fprintln(w, "Starting database...")
 	// Creating volume will not override existing volume, so we must inspect explicitly

